core/src/main/java/org/apache/brooklyn/util/core/task/ScheduledTask.java

```diff
@@ -84,12 +99,15 @@ public ScheduledTask(Map flags, Callable<Task<?>> taskFactory) {
         delay = Duration.of(elvis(flags.remove("delay"), 0));
         period = Duration.of(elvis(flags.remove("period"), null));
         maxIterations = elvis(flags.remove("maxIterations"), null);
+        Object cancelFlag = flags.remove("cancelOnException");
+        cancelOnException = cancelFlag == null || Boolean.TRUE.equals(cancelFlag);
```

Comment:
Should we do a type coercion? No strong feelings. This deep in the code, we can assume folk are using the right types.